bioptim/examples/getting_started/example_multinode_constraints.py

```diff
@@ -0,0 +1,177 @@
+"""
+This example is a trivial box that must superimpose one of its corner to a marker at the beginning of the movement and
+a the at different marker at the end of each phase. Moreover a constraint on the rotation is imposed on the cube.
+Please note that the "last" control. Extra constraints are defined: the states of the first_node of the phase 0
+are equal to the states of the first_node of the phase 2. Extra objective are defined: the states of the second node
+of the phase 0 are equal to the states of the mid node of the phase 2, the states of the middle node of the phase 0
+are equal to the states of the end node of the phase 1.
+It is designed to show how one can define a multinode constraints and objectives in a multiphase optimal control program
+"""
+
+
+import biorbd_casadi as biorbd
+from bioptim import (
+    PenaltyNode,
+    OptimalControlProgram,
+    DynamicsList,
+    DynamicsFcn,
+    ObjectiveList,
+    ObjectiveFcn,
+    ConstraintList,
+    ConstraintFcn,
+    BoundsList,
+    QAndQDotBounds,
+    InitialGuessList,
+    OdeSolver,
+    Node,
+    Solver,
+    MultinodeConstraintList,
+    MultinodeConstraintFcn,
+)
+
+
+def minimize_difference(all_pn: PenaltyNode):
+    return all_pn[0].nlp.controls.cx_end - all_pn[1].nlp.controls.cx
+
+
+def prepare_ocp(
+    biorbd_model_path: str = "models/cube.bioMod", ode_solver: OdeSolver = OdeSolver.RK4(), long_optim: bool = False
+) -> OptimalControlProgram:
+    """
+    Prepare the ocp
+
+    Parameters
+    ----------
+    biorbd_model_path: str
+        The path to the bioMod
+    ode_solver: OdeSolver
+        The ode solve to use
+    long_optim: bool
+        If the solver should solve the precise optimization (500 shooting points) or the approximate (50 points)
+
+    Returns
+    -------
+    The OptimalControlProgram ready to be solved
+    """
+
+    biorbd_model = (biorbd.Model(biorbd_model_path), biorbd.Model(biorbd_model_path), biorbd.Model(biorbd_model_path))
+
+    # Problem parameters
+    if long_optim:
+        n_shooting = (100, 300, 100)
+    else:
+        n_shooting = (20, 30, 20)
```

Comment:
Please clean what is not necessary from the copy-paste to showcase what you want to showcase

Reply:
simplified